### PR TITLE
Replace unlicensed font

### LINKED ISSUE #120: Font licensing issue (!!!)
I noticed that the font **"Apercu Pro"** is included with this package. The font files do not contain a license file aside them, which is a bit unusual. Upon further research it seems that the fonts are sold by Colophon Foundry (https://www.colophon-foundry.org/typefaces/apercu/) and from what I can tell are NOT freely available.

I do not know if this theme carries a license to distribute these fonts. If so, then this theme needs to include such a license. However I would not want theme users to suddenly get hit with copyright infringement lawsuits (it has happened to me before, and can be really destructive). Yes, something as simple as distributing an unlicensed font on your website, even unknowingly so, can wreak legal havoc.

So the 3 options are:
* Include a license which permits use and redistribution.
* Remove the font and include a more liberally licensed alternative (Source Sans Pro, or Overpass, seem like similar alternatives).
* Do not provide a font; use a web-safe/system font stack. This is a popular light-weight option, and makes it easy for theme users to override our theme choice. (Trebuchet MS is web-safe and looks loosely similar)

From the Torchbox viewpoint, Torchbox can continue using Apercu Pro on the Wagtail docs, using its own license, but will have to override whatever font is provided by this theme.

### PR DESCRIPTION
**Pull request by vsalvino**
Fixes #120 . Please read the legal issue presented there - this is not a change for the sake of design. We don't want our fellow Wagtail developers getting copyright infringement lawsuits.

* Replaced "Apercu Pro" font with "Source Sans 3", which has a permitted license for redistribution. This was chosen purely because it looks relatively similar to Apercu so shouldn't be too jarring a change visually. It is also a high quality font from Adobe so should make for a serviceable replacement.
* Updated Source Code Pro font files from TTF to WOFF2 format, which has a more efficient file size and better for web use. Also copied the license file provided with those woff2 files.
* Added documentation for customizing the fonts. Fixed the layout template to allow for such customizations.

NOTE: If someone owns a license for Apercu font, and wants to continue using it in the Wagtail (or Sphinx Wagtail Theme) docs, they should add their licensed files or CDN URLs in the relevant place (`/docs/` directory), not in the files distributed with the pip package.

Screenshot below shows customize docs, rendered using the new fonts.

image

### CODE REVIEW

**Review by Scotchester — COMMENTED** [review]
A few questions/suggestions while we wait for more opinions on Source Sans vs. system fonts.

File: docs/customizing.rst [hunk 1]
```diff
@@ -79,3 +79,38 @@ are valid here. To disable, set to ``None``
...
+   @import url("https://fonts.googleapis.com/css2?family=Roboto");
+
+   /* Main font used throughout the docs. */
+   body {
+     font-family: "Roboto", sans-serif;
+   }
+
+   /* Code snippets. */
+   pre, code, kbd, samp {
+     font-family: "Courier New", monospace;
```
[the remark this answers is not in the document]
Comment (vsalvino) [reply docs/customizing.rst:115]: I was going to put "Comic Sans" in the example :)

File: fonts/source-sans/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,96 @@
+Copyright 2010-2020 Adobe (http://www.adobe.com/), with Reserved Font Name
+'Source'. All Rights Reserved. Source is a trademark of Adobe in the United
+States and/or other countries.
```
[the remark this answers is not in the document]
Comment (vsalvino) [reply fonts/source-sans/LICENSE.txt:3]: I copied them verbatim from each's repository, where the files were obtained, as mandated.
* https://github.com/adobe-fonts/source-sans
* https://github.com/adobe-fonts/source-code-pro

**Comment by vsalvino** [comment]
Seems to be general approval/consensus from the slack thread, linking here for posterity.
https://wagtailcms.slack.com/archives/G0P7Q0G1Y/p1637595881102000

File: fonts/source-sans/LICENSE.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Scotchester) [reply fonts/source-sans/LICENSE.txt:3]: Ah, fair enough.

File: sphinx_wagtail_theme/layout.html [hunk 3]
```diff
@@ -36,6 +36,13 @@
     {%- endif %}
     <link rel="stylesheet" type="text/css" href="{{ pathto('_static/' + style, 1) }}" />
     <link rel="stylesheet" type="text/css" href="{{ pathto('_static/dist/fontawesome.css', 1) }}" />
+    {%- for css in css_files %}
```
Comment (Scotchester) [inline sphinx_wagtail_theme/layout.html:39]: Should this be `html_css_files` per L98 of `docs/customizing.rst`, or am I misunderstanding something about how this works?

```suggestion
    {%- for css in html_css_files %}
```

Comment (vsalvino) [reply sphinx_wagtail_theme/layout.html:39]: Nope, it is available in the jinja context as `css_files`. Then generally follow that naming pattern (e.g. html_theme_options becomes theme_options). See basic template which is where this code was ripped from: https://github.com/sphinx-doc/sphinx/blob/634dd395cca6c4eae2762bfbb2b6a8b1f109df00/sphinx/themes/basic/layout.html#L95-L101

Comment (Scotchester) [reply sphinx_wagtail_theme/layout.html:39]: How fascinating and weird 😆

**Review by Scotchester — APPROVED** [review]

File: sphinx_wagtail_theme/layout.html (hunk 3 above)
Comment (vsalvino) [reply sphinx_wagtail_theme/layout.html:39]: Yea sphinx internals are a bit grisly in places. I'm sure at some point I will end up refactoring the whole layout template to inherit from basic, until then it's copy-pasta for dinner.

**Pull request merged**